homeassistant/components/somfy/cover.py

```diff
@@ -0,0 +1,106 @@
+"""
+Support for Somfy Covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.somfy/
+"""
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION, \
+    ATTR_TILT_POSITION
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES, API
+
+DEPENDENCIES = ['somfy']
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy cover platform."""
+    def get_covers():
+        """Retrieve covers."""
+        from pymfy.api.devices.category import Category
+
+        categories = {Category.ROLLER_SHUTTER.value,
+                      Category.INTERIOR_BLIND.value,
+                      Category.EXTERIOR_BLIND.value}
+
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [SomfyCover(cover, hass.data[DOMAIN][API]) for cover in
+                devices if
+                categories & set(cover.categories)]
+
+    async_add_entities(await hass.async_add_job(get_covers), True)
+
+
+class SomfyCover(SomfyEntity, CoverDevice):
+    """Representation of a Somfy cover device."""
+
+    def __init__(self, device, api):
+        """Initialize the Somfy device."""
+        from pymfy.api.devices.blind import Blind
+        super().__init__(device, api)
+        self.cover = Blind(self.device, self.api)
+
+    async def async_update(self):
+        """Update the device with the latest data."""
+        from pymfy.api.devices.blind import Blind
+        await super().async_update()
+        self.cover = Blind(self.device, self.api)
```

Comment:
Is this async safe?

Reply:
Why create a new blind device on every update? This looks inefficient.

Reply:
It was why I did in a first try, only update self.device. It's done in `async_update`. Doing this, self.cover will still have its self.device targeting the old one . So it means, SomfyCover will not be aware of the new state.